nautobot/dcim/models/locations.py

```diff
@@ -87,15 +87,26 @@ def clean(self):
 
     @property
     def display(self):
-        # prefix with parent name allows location type name identity if there are the same name exists under different parents.
-        display_str = self.name
-        parent = self.parent
-        for _ in range(10):  # "10 levels iteration"
-            if parent is None:
-                break
-            display_str = f"{parent.name} → {display_str}"
-            parent = parent.parent
-        return display_str
+        """
+        Prefix with parent name allows location type name identity if there are the same name exists under different parents.
+        self.ancestors() return all the preceding nodes from top down. So if we are looking at node C and its node structure is the following:
+            A
+           /
+          B
+         /
+        C
+        This method will return "A → B → C".
+        If `self.ancestors()` throws an `ObjectDoesNotExist` exception during bulk operations, we handle it by only returning the names of the current node and the existing ancestors.
+        """
```

Comment:
Please wrap to 120 character line length, perhaps:

```suggestion
        """
        Include the parent type names as well in order to provide UI clarity.

        `self.ancestors()` returns all the preceding nodes from the top down.
        So if we are looking at node C and its node structure is the following:
            A
           /
          B
         /
        C
        This method will return "A → B → C".

        Note that `self.ancestors()` may throw an `ObjectDoesNotExist` during bulk-delete operations.
        """
```